server/src/main/java/com/epam/aidial/core/server/log/GfLogStore.java

```diff
@@ -67,9 +68,13 @@ private Void doSave(ProxyContext context) {
     private void append(ProxyContext context, LogEntry entry) throws JsonProcessingException {
         HttpServerRequest request = context.getRequest();
         HttpServerResponse response = context.getResponse();
+        ApiKeyData apiKeyData = Optional.ofNullable(context.getProxyApiKeyData())
```

Comment:
ApiKeyData could be retrieved from `context.getApiKeyData` directly if headers are unavailable in http request.

Reply:
updated